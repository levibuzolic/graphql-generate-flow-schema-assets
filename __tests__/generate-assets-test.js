const generateEnums = require('../src/generateEnums');
const generateObjectTypes = require('../src/generateObjectTypes');

let types;

beforeEach(() => {
  types = require('../schema.json').__schema.types;
});

describe('Generate assets', () => {
  it('should generate enums', () => {
    expect(generateEnums(types)).toMatchSnapshot();
  });

  it('should generate object types', () => {
    expect(generateObjectTypes(types)).toMatchSnapshot();
  });
});
