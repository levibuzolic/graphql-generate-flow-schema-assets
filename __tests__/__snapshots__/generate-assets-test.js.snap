// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate assets should generate enums 1`] = `
"// @flow

/**
 * THIS FILE IS AUTO GENERATED, DO NOT EDIT DIRECTLY.
 */

export type UserRolesEnum = 'User' | 'Admin' | 'Collaborator';

export const UserRoles: {|
  +User: 'User',
  +Admin: 'Admin',
  +Collaborator: 'Collaborator'
|} = { User: 'User', Admin: 'Admin', Collaborator: 'Collaborator' };

export type UserStatusesEnum = 'Inactive' | 'Active' | 'Suspended' | 'Banned';

export const UserStatuses: {|
  +Inactive: 'Inactive',
  +Active: 'Active',
  +Suspended: 'Suspended',
  +Banned: 'Banned'
|} = {
  Inactive: 'Inactive',
  Active: 'Active',
  Suspended: 'Suspended',
  Banned: 'Banned'
};
"
`;

exports[`Generate assets should generate object types 1`] = `
"// @flow

/**
 * THIS FILE IS AUTO GENERATED, DO NOT EDIT DIRECTLY.
 */

export type ObjectTypesEnum =
  | 'Root'
  | 'Film'
  | 'Species'
  | 'Planet'
  | 'Person'
  | 'Starship'
  | 'Vehicle';

export const ObjectTypes: {|
  +Root: 'Root',
  +Film: 'Film',
  +Species: 'Species',
  +Planet: 'Planet',
  +Person: 'Person',
  +Starship: 'Starship',
  +Vehicle: 'Vehicle'
|} = {
  Root: 'Root',
  Film: 'Film',
  Species: 'Species',
  Planet: 'Planet',
  Person: 'Person',
  Starship: 'Starship',
  Vehicle: 'Vehicle'
};
"
`;
